umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj

```diff
@@ -73,6 +75,163 @@
     science-keywords
     su/not-provided-science-keywords))
 
+(defn- distributor?
+  "Returns true if roles includes DISTRIBUTOR"
+  [roles]
+  (if (some #(= "DISTRIBUTOR" %) roles)
+    true
+    false))
+
+(defn- archiver?
+  "Returns true if roles includes ARCHIVER"
+  [roles]
+  (if (some #(= "ARCHIVER" %) roles)
+    true
+    false))
+
+(defn- add-archiver
+  "Adds ARCHIVER to roles in approriate position"
+  [roles]
+  (let [dist-index (.indexOf roles "DISTRIBUTOR")
+        dist-subvec (subvec roles 0 (inc dist-index))
+        rest-subvec (subvec roles (inc dist-index))]
+    (concat dist-subvec ["ARCHIVER"] rest-subvec)))
+
+(defn- add-distributor
+  "Adds DISTRIBUTOR to roles in approriate position"
+  [roles]
+  (let [arc-index (.indexOf roles "ARCHIVER")
+        arc-subvec (subvec roles arc-index)
+        rest-subvec (subvec roles 0 arc-index)]
+    (concat rest-subvec ["DISTRIBUTOR"] arc-subvec)))
+
+(defn- fix-archiver-distributor
+  "ISO SMAP maps DISTRIBUTOR and ARCHIVER from the same place, so anytime DISTRIBUTOR is present so must be ARCHIVER,
+   and vice versa"
+  [roles]
+  (distinct
+   (cond
+     (and (archiver? roles)
+          (distributor? roles))
+     (-> (remove #(= % "ARCHIVER") roles)
+         vec
+         add-archiver)
+     (archiver? roles)
+     (add-distributor roles)
+     (distributor? roles)
+     (add-archiver roles)
+     :default roles)))
+
+(defn- expected-data-center-roles
+  "Returns data center with :Roles modified to what is expected"
+  [data-center]
+  (let [roles (-> (:Roles data-center)
+                  fix-archiver-distributor
+                  distinct)]
+    (assoc data-center :Roles roles)))
+
+(defn- expected-iso-contact-information
+  "Returns expected contact information - 1 address, only certain contact mechanisms are mapped"
+  [contact-info url-content-type]
+  (let [contact-info (-> contact-info
+                         (update :RelatedUrls iso/expected-contact-info-related-urls)
+                         (update-in-each [:RelatedUrls] #(assoc % :URLContentType url-content-type))
+                         (update-in-each [:RelatedUrls] #(assoc % :Type "HOME PAGE"))
+                         (update :ContactMechanisms iso/expected-iso-contact-mechanisms)
+                         (update :Addresses #(seq (take 1 %))))]
+    (if (empty? (cmr.common.util/remove-nil-keys contact-info))
```

Comment:
Nit: why not just call `remove-nil-keys` at the end of the `let`? Why not require `cmr.common.util`?